### ISSUE TITLE
SRI issue with absolute paths as sources

### ISSUE THREAD

**Issue by aosdf6dd** [mozilla/http-observatory#182]
Hello!

![1481819582](https://cloud.githubusercontent.com/assets/20280659/21232819/421045a2-c2ed-11e6-93a9-b52f800e9b0a.png)

When I scan https://someurl.com (sample domain) the result looks like in the screenshot attached. The test for "subresource-integrity" lists several "external" resources that are linked. Obviously the resources listed are NOT external but from the same domain I started the scan against.

Having a closer look at it, I found some other script resources in the URL's source code of the page, which are not listed in the observatory scan result. I noticed that those were specified as relative paths. Apparently the scanner cannot recognize when absolute URLs (like https://....) are loaded which are from the same host.

It would be nice if it could recognize absolute paths pointing to the same URL, as there are many applications and frameworks that actually convert every href/src/link in an application to absolute URLs.

best regards!

**Comment by JSkier21** [comment]
I noticed this as well (getting dinged 50 points for a couple of self hosted files referenced). According to the scan, it has https: missing (site is https only). Seems to be a common issue with WordPress and jquery. Doesn't seem to consider that files like these are stored on the same domain name. 

Sample with fake site:
`//somesite.com/wp-includes/js/jquery/jquery.js`

**Comment by floatingatoll** [comment]
This is by design. Relative protocol links (href=//) are unsafe in some
scenarios and can enable HTTP downgrade attacks. Therefore, they count as
HTTP links.

Please convert all href="// links to href="https:// links to remove this
penalty.

If you’re using href=// then you’re already able to switch to href=https://,
because there is no mixed content warning when loading HTTPS resources
inside HTTP pages.

**Comment by JSkier21** [comment]
Thank you for the explanation, I see the logic there.

This works great if I'm developing something myself, but in this case I'm not. Not sure if a check to see if it is accessible over port 80 is worth it for this project, but that is what I would be interested in. 

I suppose I could also do a rewrite in Apache for these directories, instead of fixing the links on source code that is subject to change.

**Comment by scrouet** [comment]
Hi,

No progress nor target end date to solve this bug?

**Comment by Nightfirecat** [comment]
As was mentioned, this isn't a bug. I'm not really sure why it's labeled as such.

**Comment by floatingatoll** [comment]
> This works great if I'm developing something myself, but in this case I'm not. Not sure if a check to see if it is accessible over port 80 is worth it for this project, but that is what I would be interested in.

In this scenario, since you're scanning a development site, you will see the reminder that your inclusion of `src=//` links is unsafe, and as developer of that site will understand why. If your release site ends up containing those links, the penalty will correctly warn of the risk inherent, which you should then correct in your release site using whatever means you see fit.

**Comment by noneymous** [comment]
The original post was about absolute URLs to the same target not being understood as the same target. The subsequent later was about relative protocol URLs "//", which are something additional/different. 

A website (e.g. accessed via https://someurl.com) can contain HREFs in the following styles:
- /index.html (relative URL)
- https://someurl.com/index.html (absolute URL)
- //someurl.com/index.html (relative protocol URL)

**Comment by floatingatoll** [comment]
Alright.

**Comment by scrouet** [comment]
> The original post was about absolute URLs to the same target not being understood as the same target. The subsequent later was about relative protocol URLs "//", which are something additional/different.
> 
> A website (e.g. accessed via https://someurl.com) can contain HREFs in the following styles:
> 
> * /index.html (relative URL)
> * https://someurl.com/index.html (absolute URL)
> * //someurl.com/index.html (relative protocol URL)

Indeed. That is the issue we are facing with our company website build on Wordpress; we have absolute URLs automatically produced by Wordpress.
